### PR TITLE
Revise some NOTES and README files

### PR DESCRIPTION
**Pull request by mspncp**
See commit messages.

### CODE REVIEW

File: NOTES-Perl.md [hunk 1]
```diff
@@ -1,125 +1,133 @@
...
+You MUST have at least Perl version 5.10.0 installed.  This minimum
+requirement is due to our use of regexp backslash sequence \R among
+other features that didn't exist in core Perl before that version.
+
+Perl on Windows
+---------------
+
+There are a number of build targets that can be viewed as "Windows".
+Indeed, there are `VC-*` configs targeting VisualStudio C, as well as
+MinGW and Cygwin. The key recommendation is to use "matching" Perl,
```
Comment (paulidale) [inline NOTES-Perl.md:37]: to use _a_ "matching" Perl

**Review by paulidale — APPROVED** [review]
One wording suggestion.  It was preexisting, so not critical.

File: NOTES-Windows.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+     The MSYS2 version of gcc will not work correctly here.
+
+   In the MSYS2 shell do the configuration depending on the target architecture:
+
+     ./Configure mingw ...
+   or
+     ./Configure mingw64 ...
+   or
+     ./Configure ...
+   for the default architecture.
```
Comment (levitte) [inline NOTES-Windows.md:154]: I'd suggest adding a couple more spaces on the command lines, to make them verbatim.

File: NOTES-Windows.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+   Apart from that, follow the Unix / Linux instructions in INSTALL.md.
+
+ * It is also possible to build mingw[64] on Linux or Cygwin.
+
+   In this case configure with the corresponding --cross-compile-prefix= option.
+   For example
+
+     ./Configure mingw --cross-compile-prefix=i686-w64-mingw32- ...
+   or
+     ./Configure mingw64 --cross-compile-prefix=x86_64-w64-mingw32- ...
```
Comment (levitte) [inline NOTES-Windows.md:165]: I'd suggest adding a couple more spaces on the command lines, to make them verbatim.

File: README-Engine.md [hunk 4]
```diff
@@ -1,308 +1,308 @@
 ENGINES
 =======
```
Comment (levitte) [inline README-Engine.md:2]: This should probably have an early note saying the this is a deprecated feature that's replaced with providers, and that it's planned to go away.

File: NOTES-Perl.md (hunk 1 above)
Comment (mspncp) [reply NOTES-Perl.md]: I even went further and reformulated the entire sentence, removing the awkward quotes, see 087f6848504146f9cf2a52e72404efbead2eb0e6.

File: NOTES-Windows.md (hunk 2 above)
Comment (mspncp) [reply NOTES-Windows.md]: Done in c321b8eda3c81c820f02db67296df8b953bf41b9.

**Comment by mspncp** [comment]
This PR has been heavily revised. Please re-review.

**Comment by mspncp** [comment]
@mattcaswell you might be interested in commit [49a790e4cf87](https://github.com/openssl/openssl/pull/14042/commits/49a790e4cf871e848310d0d48205fa58a8e4c356).

File: README-Engine.md (hunk 4 above)
Comment (mspncp) [reply README-Engine.md]: Good point. Please take a look at [0067c99cc592](https://github.com/openssl/openssl/pull/14042/commits/0067c99cc5923ab4c4e7512fdc8fd49410cda954) for the result.

File: README-PROVIDERS.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    #include <stdio.h>
+    #include <stdlib.h>
+    
+    #include <openssl/provider.h>
+    
+    int main(void)
+    {
+        OSSL_PROVIDER *legacy;
+        OSSL_PROVIDER *deflt;
+
```
Comment (mspncp) [inline README-PROVIDERS.md:132]: Note: git complains about a lot of trailing whitespace in the empty lines of the code block. I'm not sure yet whether these lines can be left entirely empty, or whether the four spaces are required to keep the code block contiguous.

Comment (richsalz) [reply README-PROVIDERS.md:132]: You can try removing the trailing whitespace, push to your branch, and see what it looks like.  I assume you know that.

Comment (mspncp) [reply README-PROVIDERS.md:132]: Sure. Just was in a hurry to do some shopping before 8 PM. ;-)

Comment (mspncp) [reply README-PROVIDERS.md:132]: I normally just create an empty markdown file via GitHub web interface, paste my contents and switch to preview mode. No need to push anything...

Comment (mspncp) [reply README-PROVIDERS.md]: Ok, GitHub says "no problem". Whitespace cleanup pushed.

**Comment by mspncp** [comment]
> @mattcaswell you might be interested in commit [49a790e4cf87](https://github.com/openssl/openssl/pull/14042/commits/49a790e4cf871e848310d0d48205fa58a8e4c356).

I'm not decided yet how detailed the README-ENGINES should be. At least, there should be some basic introduction to the subject, giving the user as a starting point. I.e., it should point to the relevant entry points of the manual pages, and could optionally link to the wiki for additional and more current resources.

File: README-PROVIDERS.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+The FIPS Provider
+-----------------
+
+The FIPS provider contains a sub-set of the algorithm implementations available
+from the default provider, consisting of algorithms conforming to FIPS standards.
+It is intended that this provider will be FIPS140-2 validated.
+
+In some cases there may be minor behavioural differences between algorithm
+implementations in this provider compared to the equivalent algorithm in the
+default provider. This is typically in order to conform to FIPS standards.
```
Comment (paulidale) [inline README-PROVIDERS.md:65]: _typically_ seems wrong here.  The differences are always to conform to the standards.

File: README-PROVIDERS.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    [default_sect]
+    activate = 1
+    
+    [legacy_sect]
+    activate = 1
+    
+
+It is also possible to load providers programmatically. For example you can
+load the legacy provider into the default library context as shown below.
+Note that once you have explicitly loaded a provider into the library context
```
Comment (paulidale) [inline README-PROVIDERS.md:118]: I'd use _loaded a provider into **a** library context_

**Review by paulidale — APPROVED** [review]

**Comment by mspncp** [comment]
_Editorial Note: I noticed that I should rename `NOTES.WIN` to  `NOTES-WINDOWS` in the commit message of the cherry-picked commit https://github.com/openssl/openssl/pull/14042/commits/aeb6ef74e6e217d796ec4a55de1e6b771aa443f8, since the file name changed. I'll do that when merging._

**Comment by mspncp** [comment]
Merged to master, thanks for the review!

* 70f2364882 NOTES-WINDOWS: fix typo
* a0ca1eed24 Add a skeleton README-PROVIDERS file
* d507436a26 Add deprecation note to the README-ENGINES file
* 4148581eb2 Unify the markdown links to the NOTES and README files
* dc589daec8 Reformat some NOTES and README files
* 9f1fe6a950 Revise some renamings of NOTES and README files

**Pull request closed without merging**